### PR TITLE
Remove the PreviewWorkspace SolutionCrawler functionality

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
SolutionCrawler here serves no purpose.  The preview window is already only in one of two modes:

1. Roslyn-Native-Pull-Diagnostic-Tagging.  This has been how tagging has worked in roslyn since https://github.com/dotnet/roslyn/pull/63858.  In this world, diagnostics taggers just pull directly from teh diagnostic subsystem, and do not need the crawler to push results into them.
2. LSP-pull-diagnostics.  The future (and enabled for a subset of people).  In this world the preview window doesn't get any diagnostics anyways as there's no channel to feed that information back to the client.  We'll likely have to work through an LSP-native way to get this to work.  But, regardless, that won't be Solution-Crawler-Based anyways.

So yaay, we just get to ditch this.

There is still work to be done to get the pull-tagging in the preview window to do the [computation in OOP](https://github.com/dotnet/roslyn/issues/67014), not in proc.  But that's upcoming.

### CODE REVIEW

File: src/EditorFeatures/Test/Preview/PreviewWorkspaceTests.cs [hunk 1]
```diff
@@ -125,52 +125,15 @@ public async Task TestPreviewServices()
...
 
             var persistentService = previewWorkspace.Services.SolutionServices.GetPersistentStorageService();
 
             await using var storage = await persistentService.GetStorageAsync(SolutionKey.ToSolutionKey(previewWorkspace.CurrentSolution), CancellationToken.None);
             Assert.IsType<NoOpPersistentStorage>(storage);
         }
 
-        [WorkItem(923196, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/923196")]
-        [WpfFact]
-        public async Task TestPreviewDiagnostic()
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test/Preview/PreviewWorkspaceTests.cs:138]: pull tagger test is below.  this one was testing the push/solution-crawler version.

File: src/EditorFeatures/Test/Preview/PreviewWorkspaceTests.cs [hunk 2]
```diff
@@ -201,9 +164,6 @@ public async Task TestPreviewDiagnosticTaggerInPreviewPane()
 
             workspace.TryApplyChanges(workspace.CurrentSolution.WithAnalyzerReferences(new[] { DiagnosticExtensions.GetCompilerDiagnosticAnalyzerReference(LanguageNames.CSharp) }));
 
-            // set up listener to wait until diagnostic finish running
-            _ = workspace.ExportProvider.GetExportedValue<IDiagnosticService>();
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test/Preview/PreviewWorkspaceTests.cs:205]: was previously using push.  updated to pull below.

File: src/Features/LanguageServer/Protocol/Features/Diagnostics/DiagnosticAnalyzerService_IncrementalAnalyzer.cs [hunk 3]
```diff
@@ -21,17 +18,12 @@ internal partial class DiagnosticAnalyzerService : IIncrementalAnalyzerProvider
...
-
-                return NoOpIncrementalAnalyzer.Instance;
-            }
+            var analyzer = _map.GetValue(workspace, _createIncrementalAnalyzer);
 
-            return _map.GetValue(workspace, _createIncrementalAnalyzer);
+            // We rely on LSP to query us for diagnostics when things have changed and poll us for changes that might
+            // have happened to the project or closed files outside of VS. However, we still need to create the analyzer
+            // so that the map contains the analyzer to run when pull diagnostics asks.
+            return GlobalOptions.IsLspPullDiagnostics() ? NoOpIncrementalAnalyzer.Instance : analyzer;
```
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Features/Diagnostics/DiagnosticAnalyzerService_IncrementalAnalyzer.cs:26]: same logic a before.  i just rejiggered.  in both paths below we were adding hte item to the map.  i pulled that above, and then just changed the logic if the NoOp analyzer is returned or the actual instance.

**Review by dibarbet — APPROVED** [review]

**Comment by mavasani** [comment]
Merging to facilitate lightbulb performance measurements with this change.

**Pull request merged**